### DISCUSSION TITLE
Java 8 support timeline

### DISCUSSION THREAD

**Discussion by dssaggu1991** [rabbitmq/rabbitmq-jms-client#382] [Q&A] [answered]
hi guys

I wonder how long Java 8 will be supported for JMS Client 2.x ?

**Comment by lukebakken** [comment]
Based on this chart, you are living on borrowed time:

https://endoflife.date/oracle-jdk

**Comment by michaelklishin** [comment, marked as answer]
It's never all that straightforward [with Oracle products](https://www.oracle.com/ca-en/java/technologies/java-se-support-roadmap.html).

This client doesn't change much, so the answer is likely "while it is trivial to support JDK 8". Once it stops being trivial, it likely will be dropped soon enough.

There are other factors involved, for example, older JDK releases will eventually [be left behind by newer TLS versions, features, saner defaults](https://rabbitmq.com/ssl.html#tls-version-support-in-jdk-and-net).

It is a very good idea to move to JDK 17 soon. It does not make any sense that Oracle plans to support JDK 8 until after JDK 17 goes completely out of support. Very few engineering teams will consider supporting a 16 year old runtime, for the lack of newer language features alone and also because various Linux distribution will long have moved on from JDK 8 or even 17 by 2030.

**Reply by dssaggu1991** [reply]
I get that the thing is my product is based on other framework which supports java 8. So, I wonder which version of RabbitMQ I can use which works well with Java 8 and can be supported for around 2 years.

**Reply by michaelklishin** [reply]
You should use the [RabbitMQ Java client](https://rabbitmq.com/api-guide.html#jdk-versions) and/or the [RabbitMQ Stream Java client](https://rabbitmq.github.io/rabbitmq-stream-java-client/stable/htmlsingle/), both support JDK 8 right now (although it took some effort to confirm that for the stream client).

Even if we drop support for JDK 8 in a year, the current versions will still work perfectly fine with RabbitMQ 3.13 and 4.0, and likely 4.x.

For example, you can use RabbitMQ Java client 2.0 from a good decade ago with RabbitMQ 3.13 RCs. Not every feature and behavior will be available or identical but all key operations will work just like they did in 2012.

**Reply by dssaggu1991** [reply]
So, basically I see **amqp-client-5.20.0.jar** supports Java 8. But it does not have extended support end date. 

As per your message, even you drop support for JDK 8 in a year, the current versions will still work perfectly fine with RabbitMQ 3.13 and 4.0, and likely 4.x.  right ?

**Reply by michaelklishin** [reply]
Correct. We encourage you stay on the latest version as long as it supports the JDK version you must use, of course.
